File: src/test/java/com/github/k0zka/contentcompress/DotNotFilterTest.java
package com.github.k0zka.contentcompress;

import java.io.File;
import java.io.IOException;
import java.util.UUID;

import org.apache.commons.io.FileUtils;
import org.hamcrest.CoreMatchers;
import org.junit.After;
import org.junit.Assert;
import org.junit.Before;
import org.junit.Test;

public class DotNotFilterTest {

	File testDir;
	File gitDir;
	File cssDir;

	@Before
	public void setup() throws IOException {
		testDir = new File("target", "testdir-" + UUID.randomUUID());
		testDir.mkdirs();
		gitDir = new File(testDir, ".git");
		gitDir.mkdirs();
		cssDir = new File(testDir, "css");
		cssDir.mkdirs();
	}

	@After
	public void cleanup() throws IOException {
		FileUtils.deleteDirectory(testDir);
	}

	@Test
	public void accept() {
		Assert.assertThat(
				"should not accept a .something directory, it is considered a 'hidden' directory on unix-like systems",
				new DotNotFilter().accept(gitDir), CoreMatchers.is(false));
		Assert.assertThat("should accept directoryes not starting with dot", new DotNotFilter().accept(cssDir),
				CoreMatchers.is(true));
	}
}

File: src/test/kotlin/com/github/k0zka/contentcompress/DotNotFilterTest.kt
package com.github.k0zka.contentcompress

import java.io.File
import java.io.IOException
import java.util.UUID

import org.apache.commons.io.FileUtils
import org.hamcrest.CoreMatchers
import org.junit.After
import org.junit.Assert
import org.junit.Before
import org.junit.Test

class DotNotFilterTest {

	private lateinit var testDir: File
	private lateinit var gitDir: File
	private lateinit var cssDir: File

	@Before
	@Throws(IOException::class)
	fun setup() {
		testDir = File("target", "testdir-" + UUID.randomUUID())
		testDir.mkdirs()
		gitDir = File(testDir, ".git")
		gitDir.mkdirs()
		cssDir = File(testDir, "css")
		cssDir.mkdirs()
	}

	@After
	@Throws(IOException::class)
	fun cleanup() {
		FileUtils.deleteDirectory(testDir)
	}

	@Test
	fun accept() {
		Assert.assertThat(
				"should not accept a .something directory, it is considered a 'hidden' directory on unix-like systems",
				DotNotFilter().accept(gitDir), CoreMatchers.`is`(false))
		Assert.assertThat(
				"should accept directoryes not starting with dot", DotNotFilter().accept(cssDir),
				CoreMatchers.`is`(true))
	}
}
